### PR TITLE
feat: Adding snitch agent

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

 - Adding an agent who monitors the network state at every time step and stores it within itself. Once the run is over, this agent allows easy evaluation of how the simulation evolved
 - Also restructuring the Scenario setup to make it more clear how it is configured and to allow more generic control

### Testing

 - Ran tests locally
 - Integration tests passing

### Breaking Changes

 - Scenarios are restructured. Should be largely non-impacting to external use but there is a chance of issues.

### Closes

#268

### CODE REVIEW

File: vega_sim/api/data_raw.py [hunk 1]
```diff
@@ -100,30 +100,31 @@ def asset_info(
...
+    if asset_id is not None:
+        account_filter.asset_id = asset_id
+    if market_id is not None:
+        account_filter.market_ids.extend([market_id])
+    return unroll_v2_pagination(
         base_request=data_node_protos_v2.trading_data.ListAccountsRequest(
             filter=account_filter
         ),
         request_func=lambda x: data_client.ListAccounts(x).accounts,
         extraction_func=lambda res: [i.account for i in res.edges],
```
Comment (cdummett) [inline vega_sim/api/data_raw.py:125]: Not an issue with this PR, just a heads up this is going to break with vega > [v0.63.0](https://github.com/vegaprotocol/vega/releases/tag/v0.63.0) (Issue [#6275](https://github.com/vegaprotocol/vega/issues/6725) specifically)

Can't remember when we update our vega tag in market-sim but maybe worth doing as part of this PR?

**Review by cdummett — CHANGES_REQUESTED** [review]
The new `Scenario` layout looks so much cleaner!

Does have breaking changes for the `DevOpsScenario` class in `/devops/scenario.py` though..

I need to add some integration tests for these scenarios also.

```
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ python -m devops.run -s ETHUSD -n NULLCHAIN
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/caed/vega-market-sim/devops/run.py", line 96, in <module>
    main()
  File "/Users/caed/vega-market-sim/devops/run.py", line 60, in main
    scenario: Scenario = SCENARIOS[args.scenario]()
  File "/Users/caed/vega-market-sim/devops/registry.py", line 21, in <lambda>
    "ETHUSD": lambda: DevOpsScenario(
TypeError: Can't instantiate abstract class DevOpsScenario with abstract methods configure_agents, configure_environment
```

File: vega_sim/api/data_raw.py (hunk 1 above)
Comment (TomMcL) [reply vega_sim/api/data_raw.py:125]: 👌

Comment (TomMcL) [reply vega_sim/api/data_raw.py:125]: Need to attempt the wallet v2 update too so I guess I'll bundle it all up together

**Comment by TomMcL** [comment]
> The new `Scenario` layout looks so much cleaner!
> 
> Does have breaking changes for the `DevOpsScenario` class in `/devops/scenario.py` though..
> 
> I need to add some integration tests for these scenarios also.
> 
> ```
> (vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ python -m devops.run -s ETHUSD -n NULLCHAIN
> Traceback (most recent call last):
>   File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "/Users/caed/vega-market-sim/devops/run.py", line 96, in <module>
>     main()
>   File "/Users/caed/vega-market-sim/devops/run.py", line 60, in main
>     scenario: Scenario = SCENARIOS[args.scenario]()
>   File "/Users/caed/vega-market-sim/devops/registry.py", line 21, in <lambda>
>     "ETHUSD": lambda: DevOpsScenario(
> TypeError: Can't instantiate abstract class DevOpsScenario with abstract methods configure_agents, configure_environment
> ```

Agh, was careful to not break those existing methods on that but forgot the abstracts would kill it 😄. Will fix that up

**Comment by TomMcL** [comment]
@cdummett Should be fixed up now 🤞

**Review by cdummett — APPROVED** [review]
LGTM!

**Pull request merged**